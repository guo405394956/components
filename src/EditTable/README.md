# EditTable 组件



> 可编辑表格 API 基于 antd 组件中的 Table 组件，拓展了两个属性 editComponent，editConfig



## Usage



```tsx
editComponent: ()=> <Input placeholder='请输入名字'/>

editConfig:(instance: any, form: any) => ({

	rules: [{required:true, message: '请输入名字'},{validator: 自定义规则}]

})
```



## Option



| 参数                | 说明                                                         | 类型                |
| ------------------- | ------------------------------------------------------------ | ------------------- |
| editComponent       | 用于传递渲染的编辑组件,传递组件为 antd 基础  表单组件，也可自定义，需遵循 antd 组件的包装 |  funciton           |
| editConfig          | 用于传递初始值，验证规则等配置,支持 antd 表单基本校验方法， 可[查看文档](https://ant.design/components/form-cn/),支持按照 antd from 书写的自定义校验 | object              |
| columns             | 表格列配置                                                   | ColumnsItem[]       |
| data                | 数据数组                                                     | T[]                 |
| mode                | 编辑模式：全表格编辑，单行编辑                               | full \| row         |
| rowKey              | 数据自定义主键                                               | string              |
| hideOperation       | 隐藏操作按钮                                                 | boolean             |
| hideCancelConfirm   | 隐藏取消按钮二次确认                                         | boolean             |
| onChangeWithOutForm | 数据抛出方法自定义                                            | (data: T[]) => void |
| onChange            | 外界传入，表格数据抛出方法                                   | (data: T[]) => void |
| formatData4Form     | 导出数据自定义                                               | (data: T[]) => void |
| btnText             | 按钮文字自定义                                               | BtnText             |
| onSave              | mode为row，保存单条数据时触发的自定义方法                    | function            |
| onDelete            | 删除单条数据时触发的自定义方法                               | function            |
| maxNum              | 可添加的数据条数最大值                                       | number              |
| maxErrorMsg         | 表格数据达到最大值时的错误信息提示                           | string              |
| direction           | 设定数据新增在第一行或最后一行                               | top \| bottom       |
| hideDeleteConfirm   | 是否隐藏删除按钮的二次确认                                   | boolean             |



## BtnText



| 参数   | 说明               | 类型   |
| ------ | ------------------ | ------ |
| save   | 保存按钮文件自定义 | string |
| add    | 新增按钮文字自定义 | string |
| delete | 删除按钮文字自定义 | string |
| cancel | 取消按钮文字自定义 | string |
| edit   | 编辑按钮文字自定义 | string |



## ColumnsItem



| 参数          | 说明                       | 类型                          |
| ------------- | -------------------------- | ----------------------------- |
| dataIndex     | 列数据在数据项中对应的 key | string                        |
| editingStatus | 当前编辑状态               | boolean                       |
| renderCol     | 生成复杂数据的渲染函数     | Function(text, row, instance) |
| [propName]    | antd 相关属性              | [string]: any                 |



## 2021/09/17 优化内容（v0.7.16）

1. 数据添加方向自定义 新增属性 **direction**
2. 删除按钮二次确认框自定义添加 新增属性 **hideDeleteComfirm**
3. 控制表格数据条数上限 新增属性 **maxNum**，**maxErrorMsg**
4. 单条数据操作时直接发送请求 新增属性 **onSave**，**onDelete**
5. 表格数据最大值控制 新增属性 **maxNum**，**maxErrorMsg**
6. **editConfig** 属性调整，原先为传入对象，现在也支持传入方法，参数为instance和form
7. **rowKey** 属性将不再直接传入组件内部table中，仅用作传入数据主键的转化
8. 全表格编辑模式下，按钮显示调整，表格样式调整等
