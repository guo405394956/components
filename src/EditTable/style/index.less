// .ant-table-tbody > .editable-row > td {
//   padding: 20px 20px;
// }
// .editable-row {
//   .ant-form-explain {
//     position: absolute;
//   }
//   .ExtraColumn {
//     .ant-form-explain {
//       position: inherit;
//     }
//   }
// }

.editTable-full {
  td {
    padding: 0 !important;
    padding-right: 16px !important;
    border-bottom: none !important;
    vertical-align: top;
    .ant-form-item {
      margin-bottom: 20px !important;
    }
    .ant-form-item-with-help {
      margin-bottom: 0 !important;
    }
  }
}
.editTable-full:hover:not(.ant-table-expanded-row):not(.ant-table-row-selected)
  > td {
  background: #fff !important;
}
.editTable-full .operation {
  vertical-align: top;
}
.editTable-full .operation .operation-button-full {
  margin-top: 10px;
}
