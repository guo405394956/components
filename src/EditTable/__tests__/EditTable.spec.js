import {shallow} from 'enzyme';
import React from 'react';
import {Button} from 'antd';

import EditTable from '../index.tsx';
import {any} from 'prop-types';

const setup = props => {
  const columns = [
    {
      title: '姓名',
      dataIndex: 'name',
      key: 'name'
    },
    {
      title: '年龄',
      dataIndex: 'age',
      key: 'age'
    },
    {
      title: '住址',
      dataIndex: 'address',
      key: 'address'
    }
  ];
  const dataSource = [
    {
      key: '1',
      name: '胡彦斌',
      age: 32,
      address: '西湖区湖底公园1号'
    },
    {
      key: '2',
      name: '胡彦祖',
      age: 42,
      address: '西湖区湖底公园1号'
    }
  ];

  const defaultProps = {
    onChange: jest.fn()
  };

  const wrapper = shallow(
    <EditTable
      data={dataSource}
      columns={columns}
      {...Object.assign({}, defaultProps, props)}
    />
  );

  return {
    wrapper,
    props,
    dataSource,
    columns
  };
};

describe.skip('edittable base test', () => {
  it('render ', () => {
    const {wrapper, props} = setup();
    // console.log(wrapper)
    expect(wrapper.find('Table').exists()).toBe(true);
  });

  it('默认state', () => {
    const {wrapper, props} = setup({
      data: undefined
    });

    expect(wrapper.state('data')).toEqual([]);
    expect(wrapper.state('editingKey')).toBe('');
  });

  it('传值后的state', () => {
    const {wrapper, props, dataSource} = setup();

    expect(wrapper.state('data')).toEqual(dataSource);
  });

  it('render 的columns 是重新加工过的', () => {
    const {wrapper, props, dataSource, columns} = setup();

    expect(wrapper.find('Table').prop('columns')).toEqual(
      wrapper.state('columns').map(col => {
        if (!col.editComponent) {
          return col;
        }
        return {
          ...col,
          onCell: record => ({
            record,
            editConfig: col.editConfig,
            editDom: col.editComponent,
            dataIndex: col.dataIndex,
            title: col.title,
            editing: this.isEditing(record)
          })
        };
      })
    );
  });

  it.skip('render footer', () => {
    const {wrapper, props} = setup();
    expect(wrapper.find('Table').prop('footer')()).toEqual(
      <Button
        icon='plus'
        onClick={wrapper.instance().addNew}
        style={{width: '100%'}}>
        新增
      </Button>
    );
  });
});

describe('editable method', () => {
  it('renderDeleteConfirmButton 方法测试 hideDeleteConfirm=true', () => {
    const {wrapper, props} = setup();
    const instance = wrapper.instance();
    expect(
      instance.renderDeleteConfirmButton({hideDeleteConfirm: true}, {key: '1'})
    ).toMatchSnapshot();
  });
  it('renderDeleteConfirmButton 方法测试 hideDeleteConfirm=false', () => {
    const {wrapper, props} = setup();
    const instance = wrapper.instance();
    expect(
      instance.renderDeleteConfirmButton({hideDeleteConfirm: false}, {key: '1'})
    ).toMatchSnapshot();
  });
  it('renderAddAndDeleteButton 方法测试 type=add direction=bottom index=1', () => {
    const {wrapper, props} = setup({
      direction: 'bottom'
    });
    const instance = wrapper.instance();
    const type = 'add';
    const index = 1;
    wrapper.setState({
      data: [{}]
    });
    expect(instance.renderAddAndDeleteButton(type, index)).toEqual(false);
  });
  it('renderAddAndDeleteButton 方法测试 type=add direction=bottom index=0', () => {
    const {wrapper, props} = setup({
      direction: 'bottom'
    });
    const instance = wrapper.instance();
    const type = 'add';
    const index = 0;
    wrapper.setState({
      data: [{}]
    });
    expect(instance.renderAddAndDeleteButton(type, index)).toEqual(true);
  });
  it('renderAddAndDeleteButton 方法测试 type=add direction=top index=0', () => {
    const {wrapper, props} = setup({
      direction: 'top'
    });
    const instance = wrapper.instance();
    const type = 'add';
    const index = 0;
    wrapper.setState({
      data: [{}]
    });
    expect(instance.renderAddAndDeleteButton(type, index)).toEqual(true);
  });
  it('renderAddAndDeleteButton 方法测试 type=add direction=top index=1', () => {
    const {wrapper, props} = setup({
      direction: 'top'
    });
    const instance = wrapper.instance();
    const type = 'add';
    const index = 1;
    wrapper.setState({
      data: [{}]
    });
    expect(instance.renderAddAndDeleteButton(type, index)).toEqual(false);
  });
  it('renderAddAndDeleteButton 方法测试 type=delete index=1', () => {
    const {wrapper, props} = setup({
      direction: 'top'
    });
    const instance = wrapper.instance();
    const type = 'delete';
    const index = 1;
    wrapper.setState({
      data: [{}]
    });
    expect(instance.renderAddAndDeleteButton(type, index)).toEqual(false);
  });
  it('compileData 方法测试 data=undefined', () => {
    const {wrapper, props} = setup();
    const instance = wrapper.instance();
    expect(instance.compileData(undefined)).toEqual([]);
  });

  it('compileData 方法测试 data非数组', () => {
    const {wrapper, props} = setup();
    const instance = wrapper.instance();
    expect(instance.compileData({})).toEqual([]);
  });

  it('compileData 方法测试 data数组', () => {
    const {wrapper, props} = setup();
    const instance = wrapper.instance();
    expect(instance.compileData([{id: 1}])).toEqual([{id: 1, key: undefined}]);
  });

  it('compileData 方法测试 mode=full data数组长度为0', () => {
    const {wrapper, props} = setup({
      mode: 'full'
    });
    const instance = wrapper.instance();
    instance.compileData([]);
  });

  it('isEditing 默认为空，传入key为123 返回false', () => {
    const {wrapper, props} = setup();
    const record = {
      key: 123
    };
    expect(wrapper.instance().isEditing(record)).toBe(false);
  });

  it('isEditing 默认为空，传入key为空 返回true', () => {
    const {wrapper, props} = setup();
    const record = {
      key: ''
    };
    expect(wrapper.instance().isEditing(record)).toBe(true);
  });

  it('edit 方法调用,若key 为空，则返回false', () => {
    const {wrapper, props} = setup();
    const key = '';

    expect(wrapper.instance().edit(key)).toBe(undefined);
  });

  it('edit 方法调用，若key不为空，则 state 改变', () => {
    const {wrapper, props} = setup();

    const key = '123';
    wrapper.instance().edit(key);
    expect(wrapper.state('editingKey')).toBe('123');
  });

  it('delete 方法调用', () => {
    const onchangeFn = jest.fn();
    const {wrapper, props, dataSource} = setup({
      onChange: onchangeFn
    });
    const key = '1';
    let oldData = wrapper.state('data');
    wrapper.instance().delete(key);
    expect(wrapper.state('data')).toEqual(oldData.filter(c => c.key !== key));
    expect(onchangeFn.mock.calls.length).toBe(1);
  });

  it('delete 方法调用 type=delete', () => {
    const onchangeFn = jest.fn();
    const {wrapper, props, dataSource} = setup({
      onChange: onchangeFn
    });
    const instance = wrapper.instance();

    const key = '1';
    let oldData = wrapper.state('data');
    wrapper.setState({
      editingKey: '1'
    });
    expect(instance.delete(key, 'delete')).toEqual(false);

    wrapper.setState({
      editingKey: ''
    });
  });

  it('delete 方法调用 onDelete type=delete', () => {
    const onDeleteFn = jest.fn().mockImplementation((data, callback) => {
      callback(true);
    });
    const {wrapper, props, dataSource} = setup({
      onDelete: onDeleteFn
    });
    const key = '1';
    let oldData = wrapper.state('data');
    wrapper.instance().delete(key, 'delete');
    expect(wrapper.state('data')).toEqual(oldData.filter(c => c.key !== key));
    expect(onDeleteFn.mock.calls.length).toBe(1);
  });

  it('addNew 方法调用,若当前editingkey不为空则返回false', () => {
    const {wrapper, props} = setup();
    wrapper.instance().edit('1');
    expect(wrapper.instance().addNew()).toBe(false);
  });

  it('addNew 方法调用', () => {
    const {wrapper, props, dataSource} = setup();
    wrapper.instance().addNew();
    expect(wrapper.state('data').length).toBe(dataSource.length + 1);
  });

  it('addNew 方法调用 direction=bottom', () => {
    const {wrapper, props, dataSource} = setup({
      direction: 'bottom'
    });
    wrapper.instance().addNew();
    expect(wrapper.state('currentPage')).toBe(1);
  });

  it('addNew 方法调用 mode=full 校验不通过', () => {
    const {wrapper, props, dataSource} = setup({
      mode: 'full'
    });
    const instance = wrapper.instance();
    const form = {
      validateFields: callback => callback({}, {})
    };
    expect(instance.addNew({}, form)).toEqual(undefined);
  });

  it('addNew 方法调用 mode=full 校验通过', () => {
    const {wrapper, props, dataSource} = setup({
      mode: 'full'
    });
    const instance = wrapper.instance();
    const form = {
      validateFields: callback => callback(undefined, {})
    };
    instance.addNew({}, form);
  });

  it('addNew 方法调用 maxNum', () => {
    const {wrapper, props, dataSource} = setup({
      maxNum: 2
    });
    wrapper.setState({
      data: [{}, {}, {}]
    });
    expect(wrapper.instance().addNew()).toEqual(undefined);
  });

  it('renderEditConfig 方法测试 config为对象', () => {
    const {wrapper, props, dataSource} = setup();
    const instance = wrapper.instance();
    const config = {};
    const form = {};
    expect(instance.renderEditConfig(config, form)).toEqual({});
  });

  it('renderEditConfig 方法测试 config为方法', () => {
    const {wrapper, props, dataSource} = setup();
    const instance = wrapper.instance();
    const config = jest.fn().mockImplementation((data, key, form) => ({}));
    const form = {};
    expect(instance.renderEditConfig(config, form)).toEqual({});
  });

  it('renderEditConfig 方法测试 config为其他类型', () => {
    const {wrapper, props, dataSource} = setup();
    const instance = wrapper.instance();
    const config = undefined;
    const form = {};
    expect(instance.renderEditConfig(config, form)).toEqual({});
  });

  it('pageOnChange 方法测试', () => {
    const {wrapper, props, dataSource} = setup();
    const instance = wrapper.instance();
    const page = 2;
    const pageSize = 10;
    instance.pageOnChange(page, pageSize);
    expect(wrapper.state('currentPage')).toBe(2);
  });

  it('renderCell 方法测试', () => {
    const {wrapper, props, dataSource} = setup();
    const instance = wrapper.instance();
    const text = '';
    const record = {key: 1, id: 1};
    const cellConfig = {
      dataIndex: 'id',
      editComponent: jest
        .fn()
        .mockImplementation((text, record, instance, form) => (
          <div>{text}</div>
        )),
      editConfig: {}
    };
    expect(instance.renderCell(text, record, cellConfig)).toMatchSnapshot();
  });

  it.skip('cancel 方法调用', () => {
    const {wrapper, props} = setup();
    const params = {
      form: undefined,
      key: '1'
    };
    wrapper.instance().cancel(params);
    //判断 是否删除 逻辑？
  });

  it.skip('editable true branch 测试', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const render = instance.state.columns.filter(
      item => item.dataIndex === '操作'
    )[0].render;
    const record = {
      key: '1'
    };
    instance.save = jest.fn();
    instance.isEditing = jest.fn(() => true);
    const renderWrapper = render('', record);
    // dom 点击事件覆盖
    renderWrapper.props.children.props.children[0].props
      .children()
      .props.onClick();
    renderWrapper.props.children.props.children[1].props
      .children()
      .props.onConfirm();
  });

  it.skip('editable false branch 测试', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const render = instance.state.columns.filter(
      item => item.dataIndex === '操作'
    )[0].render;
    const record = {
      key: '1'
    };
    const renderWrapper = render('', record);
    // dom 点击事件覆盖
    renderWrapper.props.children.props.children[0].props.onClick();
    renderWrapper.props.children.props.children[1].props.onConfirm();
  });

  it('UNSAFE_componentWillReceiveProps 生命周期测试', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const nextprops = {
      data: [
        {
          key: '3',
          name: '胡彦组',
          age: 32,
          address: '西湖区湖底公园1号'
        }
      ]
    };
    instance.UNSAFE_componentWillReceiveProps(nextprops);
  });

  it('edit 方法测试,editingKey已存在branch', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const editingKey = '1';
    instance.state.editingKey = '5';
    const activeStatusSpy = jest.spyOn(instance, 'activeStatus');
    instance.edit(editingKey);
    expect(instance.state.editingKey).toBe('5');
    expect(activeStatusSpy).not.toHaveBeenCalledWith();
  });

  it('editColumn 方法测试', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const editingKey = 1;
    instance.editColumn(editingKey);
    expect(instance.state.editingKey).toBe(editingKey);
  });

  it('editColumn 方法测试,editingKey已存在branch', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const editingKey = '1';
    instance.state.editingKey = '5';
    instance.editColumn(editingKey);
    expect(instance.state.editingKey).not.toBe(editingKey);
  });

  // it('changeColumnEditStatus 方法测试', () => {
  //   const {wrapper} = setup();
  //   const instance = wrapper.instance();
  //   const record = {
  //     key: 1
  //   };
  //   const tdObject = {
  //     dataIndex: 'name'
  //   };
  //   const editColumnSpy = jest.spyOn(instance, 'editColumn');
  //   instance.changeColumnEditStatus(record, tdObject);
  //   expect(editColumnSpy).toHaveBeenCalledWith(record.key);
  //   expect(
  //     instance.state.columns.filter(
  //       item => item.dataIndex === tdObject.dataIndex
  //     )[0].editingStatus
  //   ).toBeTruthy();
  // });

  it('revertStatus 方法测试', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    instance.revertStatus();
    instance.state.columns.map(item => {
      expect(item.editingStatus).toBeFalsy();
    });
  });

  it('activeStatus 方法测试', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    instance.activeStatus();
    instance.state.columns.map(item => {
      expect(item.editingStatus).toBeTruthy();
    });
  });

  it('save 方法测试', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    wrapper.setState({
      data: [{key: '110'}]
    });
    const form = {
      validateFields: jest.fn(cb => {
        cb(false, null);
      }),
      getFieldInstance: jest.fn().mockReturnValue({
        validateFields: jest.fn(cb => {
          cb(false, null);
        })
      })
    };
    const key1 = '1';
    const key2 = '110';
    // 编辑保存
    instance.save(form, key1);
    // 保存新增项branch
    instance.save(form, key2);
  });

  it('save 方法测试 onSave', () => {
    const onSaveFn = jest.fn().mockImplementation((data, callback) => {
      callback(true);
    });
    const {wrapper} = setup({
      onSave: onSaveFn
    });
    const instance = wrapper.instance();
    wrapper.setState({
      data: [{key: '1'}]
    });
    const form = {
      validateFields: jest.fn(cb => {
        cb(false, null);
      }),
      getFieldInstance: jest.fn().mockReturnValue({
        validateFields: jest.fn(cb => {
          cb(false, null);
        })
      })
    };
    const key1 = '1';
    const key2 = '110';
    // 编辑保存
    instance.save(form, key1);
    // 保存新增项branch
    instance.save(form, key2);

    // expect(wrapper.state('data')).toEqual(oldData.filter(c => c.key !== key));
    expect(onSaveFn.mock.calls.length).toBe(2);
  });

  it('save 方法测试,第一个验证方法未通过branch', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const form = {
      validateFields: jest.fn(cb => {
        cb(true, null);
      })
    };
    const key = '1';
    instance.save(form, key);
  });

  it('save 方法测试,第二个验证方法未通过branch', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const form = {
      validateFields: jest.fn(cb => {
        cb(false, null);
      }),
      getFieldInstance: jest.fn().mockReturnValue({
        validateFields: jest.fn(cb => {
          cb(true, null);
        })
      })
    };
    const key = '1';
    instance.save(form, key);
  });

  it('handleChangeData 方法测试', () => {
    const onChangeFn = jest.fn();
    const {wrapper} = setup({
      onChange: onChangeFn
    });
    const instance = wrapper.instance();
    const data = [];
    instance.handleChangeData(data);
    expect(onChangeFn).toHaveBeenCalledWith([]);
  });

  it('handleChangeData 方法测试 onChangeWithOutForm!==undefined', () => {
    const onChangeFn = jest.fn();
    const formatData4FormFn = jest.fn();
    const onChangeWithOutFormFn = jest.fn();
    const {wrapper} = setup({
      onChange: onChangeFn,
      formatData4Form: formatData4FormFn,
      onChangeWithOutForm: onChangeWithOutFormFn
    });
    const instance = wrapper.instance();
    const data = [];
    instance.handleChangeData(data);
    expect(onChangeWithOutFormFn).toHaveBeenCalledWith([]);
    expect(instance.handleChangeData(data)).toEqual(false);
  });

  it('handleChangeData 方法测试 onChange===undefined', () => {
    const {wrapper} = setup({
      onChange: undefined
    });
    const instance = wrapper.instance();
    const data = [];
    expect(instance.handleChangeData(data)).toEqual(false);
  });

  it('handleChangeData 方法测试 formatData4Form !==undefined', () => {
    const onChangeFn = jest.fn();
    const formatData4FormFn = jest.fn();
    const {wrapper} = setup({
      onChange: onChangeFn,
      formatData4Form: formatData4FormFn
    });
    const instance = wrapper.instance();
    const data = [];
    instance.handleChangeData(data);
    expect(onChangeFn).toHaveBeenCalled();
    expect(formatData4FormFn).toHaveBeenCalledWith([]);
    expect(instance.handleChangeData(data)).toEqual(false);
  });

  it('cancel 方法测试, 取消table的数据行branch', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const deleteSpy = jest.spyOn(instance, 'delete');
    const revertStatusSpy = jest.spyOn(instance, 'revertStatus');
    instance.cancel({}, '1');
    expect(deleteSpy).not.toBeCalledWith('1');
    expect(instance.state.editingKey).toBe('');
    expect(revertStatusSpy).toBeCalled();
  });

  it('cancel 方法测试, 取消新增的数据行branch', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const deleteSpy = jest.spyOn(instance, 'delete');
    const revertStatusSpy = jest.spyOn(instance, 'revertStatus');
    instance.state.data.push({
      key: '3',
      name: '',
      age: '',
      address: ''
    });
    instance.cancel({}, '3');
    expect(deleteSpy).toBeCalledWith('3', 'cancel');
    expect(instance.state.editingKey).toBe('');
    expect(revertStatusSpy).toBeCalled();
  });

  it('renderDefaultConfig 方法测试 mode=row', () => {
    const {wrapper} = setup({
      mode: 'row'
    });
    const instance = wrapper.instance();

    const {footer} = instance.renderDefaultConfig();
    expect(footer()).toMatchSnapshot();
  });

  it('renderDefaultConfig 方法测试 mode=full', () => {
    const {wrapper} = setup({
      mode: 'full'
    });
    const instance = wrapper.instance();
    expect(instance.renderDefaultConfig()).toEqual({
      pagination: false
    });
  });

  it('editComponent 存在情况测试', () => {
    const props = {
      columns: [
        {
          title: '姓名',
          dataIndex: 'name',
          key: 'name545555',
          editComponent: <div></div>
        },
        {
          title: '年龄',
          dataIndex: 'age',
          key: 'age'
        },
        {
          title: '住址',
          dataIndex: 'address',
          key: 'address'
        }
      ]
    };
    setup(props);
  });
});

describe('快照测试', () => {
  it('Editable全页快照', () => {
    const {wrapper} = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
