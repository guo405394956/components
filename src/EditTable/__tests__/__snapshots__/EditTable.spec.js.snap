// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editable method renderCell 方法测试 1`] = `
<UNDEFINED>
  [Function]
</UNDEFINED>
`;

exports[`快照测试 Editable全页快照 1`] = `
<withStore(Table)
  bordered={true}
  columns={
    Array [
      Object {
        "dataIndex": "name",
        "key": "name",
        "title": "姓名",
      },
      Object {
        "dataIndex": "age",
        "key": "age",
        "title": "年龄",
      },
      Object {
        "dataIndex": "address",
        "key": "address",
        "title": "住址",
      },
      Object {
        "dataIndex": "操作",
        "render": [Function],
        "title": "操作",
      },
    ]
  }
  components={
    Object {
      "body": Object {
        "row": [Function],
      },
    }
  }
  dataSource={
    Array [
      Object {
        "address": "西湖区湖底公园1号",
        "age": 32,
        "key": undefined,
        "name": "胡彦斌",
      },
      Object {
        "address": "西湖区湖底公园1号",
        "age": 42,
        "key": undefined,
        "name": "胡彦祖",
      },
    ]
  }
  footer={[Function]}
  pagination={
    Object {
      "current": 1,
      "onChange": [Function],
      "pageSize": 10,
    }
  }
  rowClassName={[Function]}
/>
`;
